src/Compilation.zig

```diff
@@ -4562,24 +4596,116 @@ fn updateWin32Resource(comp: *Compilation, win32_resource: *Win32Resource, win32
+            defer zig_cache_tmp_dir.close();
 
-    win32_resource_prog_node.activate();
-    var child_progress_node = win32_resource_prog_node.start(rc_basename, 0);
-    child_progress_node.activate();
-    defer child_progress_node.end();
+            var output_file = zig_cache_tmp_dir.createFile(out_res_path, .{}) catch |err| {
+                return comp.failWin32Resource(win32_resource, "failed to create output file '{s}': {s}", .{ out_res_path, @errorName(err) });
+            };
+            var output_file_closed = false;
+            defer if (!output_file_closed) output_file.close();
+
+            var diagnostics = resinator.errors.Diagnostics.init(arena);
+            defer diagnostics.deinit();
+
+            var output_buffered_stream = std.io.bufferedWriter(output_file.writer());
+
+            // In .rc files, a " within a quoted string is escaped as ""
+            const fmtRcEscape = struct {
+                fn formatRcEscape(bytes: []const u8, comptime fmt: []const u8, options: std.fmt.FormatOptions, writer: anytype) !void {
+                    _ = fmt;
+                    _ = options;
+                    for (bytes) |byte| switch (byte) {
+                        '"' => try writer.writeAll("\"\""),
+                        '\\' => try writer.writeAll("\\\\"),
+                        else => try writer.writeByte(byte),
+                    };
+                }
+
+                pub fn fmtRcEscape(bytes: []const u8) std.fmt.Formatter(formatRcEscape) {
+                    return .{ .data = bytes };
+                }
+            }.fmtRcEscape;
+
+            // 1 is CREATEPROCESS_MANIFEST_RESOURCE_ID which is the default ID used for RT_MANIFEST resources
+            // 24 is RT_MANIFEST
+            const input = try std.fmt.allocPrint(arena, "1 24 \"{s}\"", .{fmtRcEscape(src_path)});
+
+            resinator.compile.compile(arena, input, output_buffered_stream.writer(), .{
+                .cwd = std.fs.cwd(),
+                .diagnostics = &diagnostics,
+                .ignore_include_env_var = true,
+                .default_code_page = .utf8,
+            }) catch |err| switch (err) {
+                error.ParseError, error.CompileError => {
+                    // Delete the output file on error
+                    output_file.close();
+                    output_file_closed = true;
+                    // Failing to delete is not really a big deal, so swallow any errors
+                    zig_cache_tmp_dir.deleteFile(out_res_path) catch {
+                        log.warn("failed to delete '{s}': {s}", .{ out_res_path, @errorName(err) });
+                    };
+                    return comp.failWin32ResourceCompile(win32_resource, input, &diagnostics, null);
+                },
+                else => |e| return e,
+            };
+
+            try output_buffered_stream.flush();
+
+            // Rename into place.
+            const digest = man.final();
```

Comment:
Since you're not adding any files to `man` that were discovered *after* running resinator, you can skip the temporary file step, call `man.final()` earlier, and write the output directly to the cache directory. The cache system will not register as a hit until `man.writeManifest()` is called below, so you don't have to worry about partial artifacts being written in case of power failure or program crashing.